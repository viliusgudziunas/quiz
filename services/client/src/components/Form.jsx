import React, { useState } from "react";

const Form = ({ formType }) => {
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleUserFormSubmit = () => {
    return;
  };

  return (
    <div>
      {formType == "Login" && <h1 className="title is-1">Log In</h1>}
      {formType == "Register" && <h1 className="title is-1">Register</h1>}
      <hr />
      <br />
      <form onSubmit={handleUserFormSubmit}>
        {formType == "Register" && (
          <div className="field">
            <input
              type="text"
              className="input is-medium"
              name="username"
              placeholder="Enter a username"
              required
              value={username}
              onChange={e => setUsername(e.target.value)}
            />
          </div>
        )}
        <div className="field">
          <input
            type="email"
            className="input is-medium"
            name="email"
            placeholder="Enter an email address"
            required
            value={email}
            onChange={e => setEmail(e.target.value)}
          />
        </div>
        <div className="field">
          <input
            type="password"
            className="input is-medium"
            name="password"
            placeholder="Enter a password"
            required
            value={password}
            onChange={e => setPassword(e.target.value)}
          />
        </div>
        <input
          type="submit"
          className="button is-primary is-medium is-fullwidth"
          value="Submit"
        />
      </form>
    </div>
  );
};

export default Form;
