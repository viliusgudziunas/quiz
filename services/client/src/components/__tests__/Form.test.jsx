import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Form from "../Form";

const testData = [
  {
    formType: "Register",
    title: "Register",
    formFields: ["username", "email", "password"]
  },
  { formType: "Login", title: "Log In", formFields: ["email", "password"] }
];

testData.forEach(el => {
  test(`${el.formType} Form renders properly`, () => {
    const component = <Form formType={el.formType} />;
    const wrapper = shallow(component);
    const h1 = wrapper.find("h1");
    expect(h1.length).toBe(1);
    expect(h1.get(0).props.children).toBe(el.title);

    const formGroup = wrapper.find(".field");
    expect(formGroup.length).toBe(el.formFields.length);
    formGroup.forEach((field, index) => {
      expect(field.get(0).props.children.props.name).toBe(el.formFields[index]);
      expect(field.get(0).props.children.props.value).toBe("");
    });

    const submitButton = wrapper.find(".button").get(0);
    expect(submitButton.props.type).toBe("submit");
    expect(submitButton.props.value).toBe("Submit");
  });

  test(`${el.formType} Form renders a snpashot properly`, () => {
    const component = <Form formType={el.formType} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
