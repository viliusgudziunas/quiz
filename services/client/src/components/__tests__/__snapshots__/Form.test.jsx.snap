// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form renders a snpashot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    Log In
  </h1>
  <hr />
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <input
        className="input is-medium"
        name="email"
        onChange={[Function]}
        placeholder="Enter an email address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        className="input is-medium"
        name="password"
        onChange={[Function]}
        placeholder="Enter a password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="button is-primary is-medium is-fullwidth"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`Register Form renders a snpashot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    Register
  </h1>
  <hr />
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <input
        className="input is-medium"
        name="username"
        onChange={[Function]}
        placeholder="Enter a username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        className="input is-medium"
        name="email"
        onChange={[Function]}
        placeholder="Enter an email address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        className="input is-medium"
        name="password"
        onChange={[Function]}
        placeholder="Enter a password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="button is-primary is-medium is-fullwidth"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
